src/main/java/ml/shifu/shifu/util/updater/BasicUpdater.java

```diff
@@ -47,13 +50,17 @@
 
     protected boolean isForSegs;
     protected List<String> segs;
+    // The column amount in original data.
+    protected int originalColumnAmount;
 
-    public BasicUpdater(ModelConfig modelConfig, int mtlIndex) throws IOException {
+    public BasicUpdater(ModelConfig modelConfig, List<ColumnConfig> columnConfigList, int mtlIndex) throws IOException {
```

Comment:
I add an argument instead of adding a new constructor, because we need this `columnConfigList` as a mandatory argument.